ndcube/mixins/ndcubeio.py

```diff
@@ -0,0 +1,257 @@
+
+
+
+    def to_hdu(self, hdu_mask='MASK', hdu_uncertainty='UNCERT'):
+        """Create a HDUList from a ImageHDU object
+
+        Parameters
+        ----------
+        hdu_mask, hdu_uncertainty : str, optional
+            If it is a string append this attribute to the HDUList as
+            `astropy.io.fits.ImageHDU` with the string as the extension name.
+            Default is `MASK` for hdu_mask, `UNCERT` for uncertainty and `None`
+            for flags.
+
+
+        Raises
+        ------
+        ValueError
+            - If `self.__mask` is set but not a `numpy.ndarray`.
+            - If `self.__uncertainty` is set but not a astropy uncertainty type
+            - If `self.__uncertainty` is set but has another unit than `self.__data`.
+
+        Returns
+        -------
+        hdulist:
+            `astropy.io.fits.HDUList`
+        """
+
+
+        # ----------------------------------HDU0------------------------------
+        # Create a copy of the meta data to avoid changing of the header of data
+        if self.__meta is not None:
+            header = fits.Header(self.__meta.copy())
+
+        else:
+            header = fits.Header()
+        # Create a FITS Safe header from the meta data
+
+        # Cross-check if the metadata is fits safe, if not then it is corrected
+        for k, v in header.items():
+            _insert_in_metadata_fits_safe(header, k, v)
+
+        if self.__wcs:
+            # Create a header for a given wcs object
+            # Hard-Coded relax parameter to write all
+            # recognized informal extensions of the WCS standard.
+            wcs_header = self.__wcs.to_header(relax=True)
+            header.extend(wcs_header, useblanks=False, update=True)
+
+        # Create a FITS header for storing missing_axis
+        # MISSING0 : 1 if axis 1 is missing
+        # other keywords are skipped, if the axis is present
+        if self.__missing_axis:
+            header0 = fits.Header()
+            for index, axis in enumerate(self.__missing_axis):
+                if axis:
+                    header0[MISSING.format(index)] = 1
+                else:
+                    continue
+
+            for k, v in header0.items():
+                _insert_in_metadata_fits_safe(header0, k, v)
+
+            header.extend(header0, useblanks=False, update=True)
+
+
+        # PrimaryHDU list contains only meta, missing_axis and wcs as headers, no data
+        hdus = [fits.PrimaryHDU(data=None, header=header)]
+
+        #------------------------------HDU0----------------------------------
+
+        #------------------------------HDU1----------------------------------
+        # Header for unit
+        if self.__unit:
+            header_unit = fits.Header()
+            if self.__unit is not u.dimensionless_unscaled:
+                header_unit['bunit'] = self.__unit.to_string()
+            hdus.append(fits.ImageHDU(self.__data, header_unit, name='UNIT'))
```

Comment:
Why is the name of this hdu ```"UNIT"``` when it appears to also contain the data?  Should it be called ```"DATA"```?

Reply:
Similar comments for the names of other hdus.

Reply:
Sure, do you want any specific names for `hdu_uncertainty` ?

Reply:
UNCERTAINTY, I guess. Or UNCERT if a shorter name is needed.